### DISCUSSION TITLE
Why data_transfer_size and max_spdm_msg_size must be equal when chunk capabilities are disabled?

### DISCUSSION THREAD

**Discussion by ssunder-xilinx** [DMTF/libspdm#2427] [Q&A] [answered]
May I know why the responder generating the SPDM_ERROR_CODE_INVALID_REQUEST error when the chunk capabilities are disabled and  data_transfer_size and max_spdm_msg_size are not equal?
The DMTF specification says that the data_transfer_size should be less than max_spdm_msg_size. Also, there is already a check for this but why do we need an additional check?
This is the code snippet in question,

external/product/libspdm/library/spdm_responder_lib/libspdm_rsp_capabilities.c

```
if (spdm_request->header.spdm_version >= SPDM_MESSAGE_VERSION_12) {
        if ((spdm_request->data_transfer_size < SPDM_MIN_DATA_TRANSFER_SIZE_VERSION_12) ||
            (spdm_request->data_transfer_size > spdm_request->max_spdm_msg_size)) {
            return libspdm_generate_error_response(spdm_context,
                                                   SPDM_ERROR_CODE_INVALID_REQUEST, 0,
                                                   response_size, response);
        }
        if (((spdm_request->flags & SPDM_GET_CAPABILITIES_REQUEST_FLAGS_CHUNK_CAP) == 0) &&
            (spdm_request->data_transfer_size != spdm_request->max_spdm_msg_size)) {
            return libspdm_generate_error_response(spdm_context,
                                                   SPDM_ERROR_CODE_INVALID_REQUEST, 0,
                                                   response_size, response);
        }
    }
```

Thanks,
Shyam

**Comment by steven-bellock** [comment, marked as answer]
This was clarified in 1.3 and will eventually be clarified in 1.2.2 when that gets published.
![image](https://github.com/DMTF/libspdm/assets/8474235/d871fc4e-e715-4178-ac27-af0f7d5220a5)

On the flip-side, if the Requester does not support chunking then what does it mean to have `DataTransferSize != MaxSPDMmsgSize`? `DataTransferSize` is the size of the transmission buffer. `MaxSPDMmsgSize` is the size of the backing buffer into which chunks are re-assembled. If chunking is not supported then the two buffers, and their sizes, are the same.

**Reply by ssunder-xilinx** [reply]
Thanks @steven-bellock
